### ISSUE TITLE
How to setup a local server for realtime streaming inferences?

### ISSUE THREAD

**Issue by deephog** [triton-inference-server/server#5089]
Hi,

I'm very new to Triton but I did all the searches I can to find the answers for my question, but there was no luck.

My use case is, we would like to setup an inference server only locally for our company so we can better concentrate our computing resources. But sometimes, we have streaming applications that needs the server be active and ready to respond (with only a specific model though), so we can save the overhead for connection establishment, model loading and initialization and etc. 

We know there will be communication overhead which is fine, and it is nothing compares to the initialization cost. The streaming process could be up indefinitely long, we capture data in real-time, and it will be fed to the inference server for results. The current "stream" API I found in the example requires a starting and ending ID, which we may not be able to provide.

Basically, we would like to send the server a command to initialize itself, keep the ready state and process the data that keeping coming in, and shut itself down or go back to semi-ready state only when the client tells it to do so. Is such scenario possible?

**Comment by jbkyang-nvi** [comment]
Hi, if you want to serve the inference server only locally you can still use a server/client format _or_ you can use the CAPI([In process API](https://github.com/triton-inference-server/server/blob/main/docs/customization_guide/inference_protocols.md#in-process-triton-server-api)) for speed.

Do you need to warmup the model? The model if it's loaded should be ready to serve the requests. Can you explain why your client cannot send start/ending id? It is fine the request doesn't end as long as the ending is specified by the `CONTROL_SEQUENCE_END` parameter.

cc: @tanmayv25  for verification

**Comment by deephog** [comment]
> Hi, if you want to serve the inference server only locally you can still use a server/client format _or_ you can use the CAPI([In process API](https://github.com/triton-inference-server/server/blob/main/docs/customization_guide/inference_protocols.md#in-process-triton-server-api)) for speed.
> 
> Do you need to warmup the model? The model if it's loaded should be ready to serve the requests. Can you explain why your client cannot send start/ending id? It is fine the request doesn't end as long as the ending is specified by the `CONTROL_SEQUENCE_END` parameter.
> 
> cc: @tanmayv25 for verification

Thank you for the insights provided! I will link this thread to the other question I asked and eventually they narrow down to the same answers from different perspectives. #5095 

one more question if convenient,  do you have a simple example somewhere that explains how to send and process sequential data stream (most example talk about sequential batching, which is definitely not what I need.) ? 

here is my very simple test code:

```
import requests
import time
import numpy as np



if __name__ == "__main__":

    input_data = np.random.rand(1, 6, 320, 640)
    request_data = {
    "inputs": [{
        "name": "input",
        "shape": [1, 6, 320, 640],
        "datatype": "FP32",
        "data": input_data.tolist()
    }],
    "outputs": [{"name": "output"}]
}
    for i in range(100):
        start = time.time()
        res = requests.post(url="http://localhost:8000/v2/models/stereo/versions/1/infer",json=request_data).json()
        print("time: ", time.time()-start)
```

which needs exact same amount of very long time (6 seconds compare to 60ms the model itself runs) no matter how many iterations I run. Which means the server releases the resources after each iteration and initialize itself again when next data piece comes. So warm-up aside, if I don't care about the speed of the first inference, I just want the model take the following sequence in a fast responding way without reinitialize itself, what setting should I use? Or in what format should I send the data?

Thanks!

**Comment by jbkyang-nvi** [comment]
I don't know if I answered your original question here. If you are not looking into sequential scheduled modes (stateful models). You can just use grpc streaming and go with default batching. You can read more about [batching strategies](https://github.com/triton-inference-server/server/blob/main/docs/user_guide/architecture.md#models-and-schedulers) here. 

The [README](https://github.com/triton-inference-server/client#triton-client-libraries-and-examples) highlights some examples we have for example the [image client](https://github.com/triton-inference-server/client#image-classification-example) has [options](https://github.com/triton-inference-server/client/blob/4bb65af5c8affadeee954e358ebead72717aebe2/src/python/examples/grpc_image_client.py#L315) for using `--streaming`.

For your model specific issue lets keep it to the same issue thread [here](https://github.com/triton-inference-server/server/issues/5095)

**Comment by jbkyang-nvi** [comment]
Closing this issue due to lack of activity. If this issue needs follow-up, please let us know and we can reopen it for you.
